app/score_calculator/divide/general_shape.py

```diff
@@ -30,19 +62,32 @@ def create_from_hand(hand: Hand) -> BlockDivisionState:
         )
 
 
-GENERAL_SHAPE_SIZE: Final[int] = 5
-TRIPLET_SIZE: Final[int] = 3
-PAIR_SIZE: Final[int] = 2
-SEQUENCE_MAX_START_POINT: Final[int] = 7
-FULLY_HAND_SIZE: Final[int] = 14
+def divide_general_shape_knitted_sub(hand: Hand) -> list[list[Block]]:
+    has_knitted_blocks: bool
+    parsed_hands: list[list[Block]] = []
+
+    for one_case in KNITTED_CASE:
+        has_knitted_blocks = True
+        for tile in one_case:
+            if hand.tiles[tile] < 1:
+                has_knitted_blocks = False
+                break
+        if not has_knitted_blocks:
+            continue
+        new_hand = deepcopy(hand)
+        for i in range(0, len(one_case), KNITTED_SIZE):
+            new_hand.call_blocks.append(
+                Block(BlockType.KNITTED, one_case[i], is_opened=False),
+            )

```

Comment:
i를 직접사용하지 않고 인덱스로만 사용하는거면 가능한한 i로 iterate하는게 아니라 value로 iterate하는게 좋습니다.
```suggestion
        for block in one_case[::KNITTED_SIZE]:
            new_hand.call_blocks.append(
                Block(BlockType.KNITTED, block, is_opened=False),
            )

```
```suggestion
    new_hand.call_blocks.extend(Block(BlockType.KNITTED, block, is_opened=False) for block in one_case[::KNITTED_SIZE])

```